### PR TITLE
[ADD] product_lifecycle: add module

### PR DESCRIPTION
**Pull request by GavyMG**
This module applies a part of the product lifecycle management concept by 
adding a status field and replacement product fields that will be used as 
alternatives in sales operations when a product is discontinued.

### CODE REVIEW

File: product_lifecycle/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        "views/sale_order_views.xml",
+        "views/assets.xml",
+    ],
+    "demo": [
+        "demo/replacement_product_demo.xml",
+    ],
+    "qweb": [
+        "static/src/xml/replacement_template.xml",
+    ],
+    "installable": True,
```
Comment (luisg123v) [inline product_lifecycle/__manifest__.py:26]: Could you add my user as maintainer, please?

**Review by luisg123v — CHANGES_REQUESTED** [review]
Since this is not a migrated module but a new one, squashing will be required.

File: product_lifecycle/__manifest__.py (hunk 1 above)
Comment (GavyMG) [reply product_lifecycle/__manifest__.py:26]: Done. Could you check this MR?

**Comment by lmignon** [comment]
@GavyMG  Part of the proposed logic is already provided  by https://github.com/OCA/product-attribute/tree/12.0/product_state Would it be possible to base your new addon on the first one?

**Pull request closed without merging**